integration/env/env.go

```diff
@@ -12,106 +12,110 @@ import (
-	// TEST_DIR env var.
-	EnvVarTestDir = "TEST_DIR"
-	// EnvVaultToken is the process environment variable representing the
-	// VAULT_TOKEN env var.
-	EnvVaultToken = "VAULT_TOKEN"
-	// EnvVarVersionBundleVersion is the process environment variable representing
-	// the VERSION_BUNDLE_VERSION env var.
-	EnvVarVersionBundleVersion = "VERSION_BUNDLE_VERSION"
-)
-
 var (
-	circleSHA            string
-	commonDomain         string
-	guestAWSArn          string
-	testedVersion        string
-	testDir              string
-	vaultToken           string
-	versionBundleVersion string
+	awsAPIHostedZoneGuest     string
+	awsIngressHostedZoneGuest string
+	awsRegion                 string
+	awsRouteTable0            string
+	awsRouteTable1            string
+	circleSHA                 string
+	clusterName               string
+	commonDomain              string
+	guestAWSARN               string
+	guestAWSAccessKeyID       string
+	guestAWSAccessKeySecret   string
+	guestAWSAccessKeyToken    string
+	hostAWSAccessKeyID        string
+	hostAWSAccessKeySecret    string
+	hostAWSAccessKeyToken     string
+	idRSAPub                  string
+	testedVersion             string
+	testDir                   string
+	registryPullSecret        string
+	vaultToken                string
+	versionBundleVersion      string
 )
 
 func init() {
-	circleSHA = os.Getenv(EnvVarCircleSHA)
-	if circleSHA == "" {
-		panic(fmt.Sprintf("env var '%s' must not be empty", EnvVarCircleSHA))
+	getenv := func(key string, value *string) {
+		*value = os.Getenv(key)
+		if *value == "" {
+			panic(fmt.Sprintf("env var '%s' must not be empty", key))
+		}
 	}
 
-	guestAWSArn = os.Getenv(EnvVarGuestAWSArn)
-	if guestAWSArn == "" {
-		panic(fmt.Sprintf("env var '%s' must not be empty", EnvVarGuestAWSArn))
-	}
+	var blackHole string
+
+	getenv("AWS_API_HOSTED_ZONE_GUEST", &awsAPIHostedZoneGuest)
+	getenv("AWS_INGRESS_HOSTED_ZONE_GUEST", &awsIngressHostedZoneGuest)
+	getenv("AWS_REGION", &awsRegion)
+	getenv("AWS_ROUTE_TABLE_0", &awsRouteTable0)
+	getenv("AWS_ROUTE_TABLE_1", &awsRouteTable1)
+	getenv("CIRCLE_SHA1", &circleSHA)
+	// TODO rename to CLUSTER_ID. Note this also had to be changed in the
+	// framework package of e2e-harness.
+	getenv("CLUSTER_NAME", &clusterName)
+	getenv("COMMON_DOMAIN", &commonDomain)
+	getenv("GITHUB_BOT_TOKEN", &blackHole)
+	getenv("GUEST_AWS_ARN", &guestAWSARN)
+	getenv("GUEST_AWS_ACCESS_KEY_ID", &guestAWSAccessKeyID)
+	getenv("GUEST_AWS_SECRET_ACCESS_KEY", &guestAWSAccessKeySecret)
+	getenv("GUEST_AWS_SESSION_TOKEN", &guestAWSAccessKeyToken)
+	getenv("HOST_AWS_ACCESS_KEY_ID", &hostAWSAccessKeyID)
+	getenv("HOST_AWS_SECRET_ACCESS_KEY", &hostAWSAccessKeySecret)
+	getenv("HOST_AWS_SESSION_TOKEN", &hostAWSAccessKeyToken)
+	getenv("IDRSA_PUB", &idRSAPub)
+	getenv("TESTED_VERSION", &testedVersion)
+	getenv("TEST_DIR", &testDir)
+	getenv("REGISTRY_PULL_SECRET", &registryPullSecret)
+	getenv("VAULT_TOKEN", &vaultToken)
```

Comment:
I was so annoyed because of jumping around with those consts that I just did it like that. You can see this list here is ~3 times longer now.

Reply:
- We tried to differentiate between common and provider specific env vars. Check for instance the `azure-operator`'s env package. My idea here was to have a common env package in `e2etests` but we should separate the code already to make the move more easy. 
- I think the approach how it was was just fine. I cannot comprehend why this should be better or why you changed it. I am rather against the new style. It is how it was in all other projects as well. May you want to elaborate a bit more on the problems you have so we can think how we can address your personal issues.

Reply:
+1 on separating common and provider. We need to be able to easily run e2e tests that don't have a guest cluster.

We also want to split logic in e2e-harness to make it more composable. See https://github.com/giantswarm/giantswarm/issues/3878 for the issue created by @MarcelMue. The refactoring going on is great and needed but we need to keep it aligned IMO.

cc @indyfree

Reply:
I'll refactor it back. I wouldn't have an issue when adding a sing var. Copying common.go from azure-operator will make things easier.

Reply:
Prime example why I think current approach is inconvenient: https://github.com/giantswarm/aws-operator/pull/1104/commits/23493cafb8fca54efb01695c7a3c066bb1ec49dd